### PR TITLE
Libs 351/headerbar apps menu

### PR DESCRIPTION
**Pull request by d-rita**
Implements [351](https://dhis2.atlassian.net/browse/LIBS-351)

---

### Description

_text_

---

### Known issues

-   [ ] _issue_

---

### Checklist

-   [ ] API docs are generated
-   [ ] Tests were added
-   [ ] Storybook demos were added

_All points above should be relevant for feature PRs. For bugfixes, some points might not be relevant. In that case, just check them anyway to signal the work is done._

---

### Screenshots

_supporting text_

### CODE REVIEW

File: components/header-bar/src/command-palette/hooks/use-navigation.js [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+    } = useCommandPaletteContext()
+
+    const goToDefaultView = useCallback(() => {
+        setFilter('')
+        setCurrentView('home')
+        setActiveSection('grid')
+        setHighlightedIndex(0)
+    }, [setActiveSection, setCurrentView, setFilter, setHighlightedIndex])
+
+    const handleListViewNavigation = useCallback(
```
Comment (kabaros) [inline components/header-bar/src/command-palette/hooks/use-navigation.js:26]: It'd be good to have a test for arrow navigation (focusing on edge cases, when list has only one item, when you're at the end of the list etc..)

File: components/header-bar/src/command-palette/hooks/use-navigation.js [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+        (event) => {
+            const lastIndex = itemsArray.length - 1
+            switch (event.key) {
+                case 'ArrowDown':
+                    setHighlightedIndex(
```
Comment (kabaros) [inline components/header-bar/src/command-palette/hooks/use-navigation.js:31]: are we sure we don't need preventDefault for ArrowDown and Up? (maybe ArrowDown scrolls the page?)

**Review by kabaros — APPROVED** [review]
lgtm - it'd be good to add tests for navigation with arrow keys (and decide where this will be merged until the backend for shortcuts and commands is implemented)

**Comment by cooper-joe** [comment]
I noticed that the highlight position is carried over from a unfiltered state to the filtered state. 
Example:
1. Open command palette
2. Move highlight to position 3 (the third app in the grid)
3. Start typing to enter the filtered state
Expectation: the highlight should be on the first item in the filtered state list
Current outcome: the highlight is on the third item in the filtered state list (and is potentially hidden if there are < 3 items).

**Review by amcgee — APPROVED** [review]

**Comment by amcgee** [comment]
🎉 🎉 🎉

**Comment by amcgee** [comment]
Merging this to a feature branch 👍
